sparse_strips/vello_common/src/glyph.rs

```diff
@@ -0,0 +1,190 @@
+// Copyright 2025 the Vello Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Processing and drawing glyphs.
+
+use crate::peniko::Font;
+use skrifa::instance::{NormalizedCoord, Size};
+use skrifa::outline::DrawSettings;
+use skrifa::{
+    GlyphId, MetadataProvider,
+    outline::{HintingInstance, HintingOptions, OutlinePen},
+};
+use vello_api::kurbo::{Affine, BezPath, Vec2};
+
+pub use vello_api::glyph::*;
+
+/// A glyph prepared for rendering.
+#[derive(Debug)]
+pub enum PreparedGlyph {
+    /// A glyph defined by its outline.
+    Outline(OutlineGlyph),
+    // TODO: Image and Colr variants.
+}
+
+/// A glyph defined by a path (its outline) and a local transform.
+#[derive(Debug)]
+pub struct OutlineGlyph {
+    /// The path of the glyph.
+    pub path: BezPath,
+    /// The local transform of the glyph.
+    pub local_transform: Affine,
+}
+
+/// Trait for types that can render glyphs.
+pub trait GlyphRenderer {
+    /// Fill glyphs with the current paint and fill rule.
+    fn fill_glyphs(&mut self, glyphs: impl Iterator<Item = PreparedGlyph>);
+
+    /// Stroke glyphs with the current paint and stroke settings.
+    fn stroke_glyphs(&mut self, glyphs: impl Iterator<Item = PreparedGlyph>);
+}
+
+/// A builder for configuring and drawing glyphs.
+#[derive(Debug)]
+pub struct GlyphRunBuilder<'a, T: GlyphRenderer + 'a> {
+    run: GlyphRun,
+    renderer: &'a mut T,
+}
+
+impl<'a, T: GlyphRenderer + 'a> GlyphRunBuilder<'a, T> {
+    /// Creates a new builder for drawing glyphs.
+    pub fn new(font: Font, renderer: &'a mut T) -> Self {
+        Self {
+            run: GlyphRun {
+                font,
+                font_size: 16.0,
+                glyph_transform: None,
+                hint: true,
+                normalized_coords: Vec::new(),
+            },
+            renderer,
+        }
+    }
+
+    /// Set the font size in pixels per em.
+    pub fn font_size(mut self, size: f32) -> Self {
+        self.run.font_size = size;
+        self
+    }
+
+    /// Set the per-glyph transform. Can be used to apply skew to simulate italic text.
+    pub fn glyph_transform(mut self, transform: Affine) -> Self {
+        self.run.glyph_transform = Some(transform);
+        self
+    }
+
+    /// Set whether font hinting is enabled.
+    pub fn hint(mut self, hint: bool) -> Self {
+        self.run.hint = hint;
+        self
+    }
+
+    /// Set normalized variation coordinates for variable fonts.
+    pub fn normalized_coords(mut self, coords: Vec<NormalizedCoord>) -> Self {
```

Comment:
This type should be as in #747

Reply:
Interesting about this taking in a `Vec`. I don't fully hate it - it's honest about the allocations. But it would also probably be better to re-use the same allocation instead (or get a lifetime from a passed in slice)

Reply:
Great idea! I've stopped exposing Skrifa API and used a normalised coords slice in https://github.com/linebender/vello/pull/883/commits/db226ef37898591d6704af2a5c3c38c1652f26c3